docs/source/getting-started/testing.rst

```diff
@@ -18,7 +18,9 @@ Of course there are more to choose from, but this is one combination that works.
 To install Puppeteer and some other testing utilities we are going to use,
 run the following command in the ``ui`` directory::
 
-    npm add --only=dev puppeteer@18.2.1 wait-on @types/jest @types/node @types/puppeteer @types/wait-on
+    npm i --save-dev puppeteer@~18.1.0 wait-on@~6.0.1 @types/jest@~29.2.3 @types/node@~18.11.9 @types/puppeteer@~7.0.4 @types/wait-on@~5.3.1
```

Comment:
Being a bit more explicit here because after working on this on and off for the past couple weeks I have lost all faith in NPM minor version compatibility (`^`). I really hope patch-level compatibility (`~`) is going to work out better; I couldn't find a way to pin exact versions through the CLI.